File: js/geese.js
function gooseFilter (birds) {
  const geese = ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"];

  let result = [];
  for (bird of birds) {
    if ( !(checkAvailability( geese, bird )) )
      result.push( bird );
  }
  return result;
};

function checkAvailability(arr, val) { // Checking whether a value exists in an array
  return arr.some(function(arrVal) {
    return val === arrVal;
  });
}
console.log(gooseFilter(["Mallard", "Hook Bill", "African", "Crested", "Pilgrim", "Toulouse", "Blue Swedish"]));

// Autres solutions intéressantes (CodeWars) :

// function gooseFilter (birds) {
//   var geese = ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"];
//   return birds.filter(b => !geese.includes(b));
// };

// function gooseFilter (birds) {
//   var geese = ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"];
//   return birds.filter( bird => geese.indexOf(bird) < 0 );
// };

// const GEESE = ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"];
// const gooseFilter = birds => birds.filter(bird => !GEESE.includes(bird));

// function gooseFilter (birds) {
//   var geese = ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"];
//   return gooseFilter.pop(geese);
//   // return an array containing all of the strings in the input array except those that match strings in geese
// };

File: lib/geese.rb
# Write a function, gooseFilter/goose_filter/GooseFilter, that takes an array of strings as an argument and returns a filtered array containing the same elements but with the 'geese' removed.

# The geese are any strings in the following array, which is pre-populated in your solution:

# def goose_filter (birds)
#   geese = ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"]

#   birds.select {|bird| !geese.include?(bird) }
# end

def goose_filter (birds)
  geese = ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"]

  birds.reject {|bird| geese.any? bird }
end


# autres solutions

# def goose_filter (birds)
#   geese = ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"]
#   birds - geese
# end

# def goose_filter (birds)
#   geese = ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"]
#   birds.reject { |bird| geese.include?(bird) }
# end

# def goose_filter (birds)
#   geese = ["African", "Roman Tufted", "Toulouse", "Pilgrim", "Steinbacher"]
#   return birds.delete_if { |is_goose| geese.include? is_goose }
# end
